tableaudocumentapi/datasource.py

```diff
@@ -116,6 +127,20 @@ def from_file(cls, filename):
         dsxml = xml_open(filename, cls.__name__.lower()).getroot()
         return cls(dsxml, filename)
 
+    @classmethod
+    def from_scratch(cls, caption, connections):
```

Comment:
Yeah, it'd be what you call, I just changed it to from_connections when I fixed a bug in my next commit